processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -378,21 +380,33 @@ public boolean isAssignableTo(Type other) {
                     modifiableGetters.put( Executables.getPropertyName( getter ), getter );
                 }
             }
-            getters = Collections.unmodifiableMap( modifiableGetters );
+
+            //TODO decide if we want to have a mechanism for turning on/off direct field mapping
```

Comment:
It is already like that. For the property read accessors, first I go through the "getter" methods and then through the fields. A field is added only if a "getter" could not be found.

For the property write accessors it is similar. The fields are last checked, they are added at the end of the `getAlternativeTargetAccessors`. I think I am correctly handling the the strategy there. 

Maybe it would be good to add tests for the specific classes that we have